app/src/main/java/com/example/myapplication/ThirdFragment.java

```diff
@@ -101,30 +102,29 @@ public void onCheckedChanged(CompoundButton buttonView, boolean isChecked)  {
                 setSwitchColor(binding.switch3);
             }
         });
+        binding.toSecondFragment.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                NavHostFragment.findNavController(ThirdFragment.this)
+                        .navigate(R.id.action_ThirdFragment_to_SecondFragment);
+                //getting the value of variables when switching to teleop from auto
+                switch1.isChecked();
```

Comment:
Almost there, however at the moment when you call these variables you're temoparily getting the value but not storing them anywhere so the moment the code rins these lines the values are lost. Assign them to a variabile and you should be good

Reply:
fixed